### ISSUE TITLE
Any interest in updating this project?

### ISSUE THREAD

**Issue by scohen** [appodeal/clickhousex#6]
Hi,
I have a fork of Clickhousex where I updated this project to use the latest Ecto / DBConnection. I also fixed some bugs and added more thorough type handling.  I also added a lot of tests.

Additionally, I have a branch that uses different output formats from Clickhouse that should make for more efficient queries.

Would you be interested in a PR?

**Comment by santaux** [comment]
Hi @scohen,
Yeah, sure, we are interested in any updates you can make :)

**Comment by scohen** [comment]
Alright. I also made [this pr](https://github.com/scohen/clickhousex/pull/1/files) that switches the response format to binary and handles all of the types that clickhouse supports. It also allows for setting the request and response types via codecs. I noticed there was a bit of conversion from the clickhouse JSON format into the list of tuples format that ecto wants, and clickhouse supports the JSONCompact format, which is a direct mapping to what ecto requires, so that should be more efficient (though, my testing didn't indicate a bottleneck in decoding queries). 

If you're interested, I'll push that here as well.
